CRM/Core/SelectValues.php

```diff
@@ -1174,27 +1174,18 @@ public static function getQuicksearchOptions(): array {
         'label' => ts('Job Title'),
       ],
     ];
-    $custom = civicrm_api4('CustomField', 'get', [
-      'checkPermissions' => FALSE,
-      'select' => ['id', 'name', 'label', 'custom_group_id.name', 'custom_group_id.title', 'option_group_id'],
-      'where' => [
-        ['custom_group_id.extends', 'IN', array_merge(['Contact'], CRM_Contact_BAO_ContactType::basicTypes())],
-        ['data_type', 'NOT IN', ['ContactReference', 'Date', 'File']],
-        ['custom_group_id.is_active', '=', TRUE],
-        ['is_active', '=', TRUE],
-        ['is_searchable', '=', TRUE],
-      ],
-      'orderBy' => [
-        'custom_group_id.weight' => 'ASC',
-        'weight' => 'ASC',
-      ],
-    ]);
-    foreach ($custom as $field) {
-      $options[] = [
-        'key' => $field['custom_group_id.name'] . '.' . $field['name'] . ($field['option_group_id'] ? ':label' : ''),
-        'label' => $field['custom_group_id.title'] . ': ' . $field['label'],
-        'adv_search_legacy' => 'custom_' . $field['id'],
-      ];
+    $customGroups = CRM_Core_BAO_CustomGroup::getAll(['extends' => 'Contact', 'is_active' => TRUE], CRM_Core_Permission::VIEW);
```

Comment:
@colemanw will this cause the custom fields to be ordered differently?

Reply:
The order isn't really important. But the new function orders by weight which is correct.